### PR TITLE
Fixed legacy dashboard patch to enable grade export

### PR DESCRIPTION
**Pull request by gsidebo**
#### What are the relevant tickets?

Closes mitocw#74

#### What's this PR do?

Adds legacy grade export functionality back into the instructor dashboard (was removed after the dogwood releases)

#### Where should the reviewer start?

Probably `lms/templates/instructor/instructor_dashboard_2/grade_export.html` and `lms/static/js/instructor_dashboard/grade_export.js`.

#### How should this be manually tested?

- Set the `ENABLE_INSTRUCTOR_GRADE_EXPORT` feature to `True`
- Set a `REMOTE_GRADEBOOK_URL` feature value - you can ask me or Peter for the proper setting here
- Navigate to the `Export Grades` tab for some course (eg: `http://localhost:8000/courses/course-v1:edX+DemoX+Demo_Course/instructor#view-grade_export` - make sure you have staff/admin privileges)
- Click the buttons and make sure they work

#### Any background context you want to provide?

1. This PR was originally opened here (https://github.com/mitodl/edx-platform/pull/7), but that was closed in favor of this one due to it being built off of the wrong branch. @blarghmatey created and pushed a patch of those changes on this branch, so this diff is effectively just a correction to that patch.
1. This grade exporting functionality last existed in the dogwood releases (eg: [dogwood.3-3](https://github.com/mitodl/edx-platform/tree/mitx/dogwood.3-3)) in something called the legacy instructor dashboard. As such there is a lot of functionality that was lifted out of the codebase as it existed then. I made several changes to match the patterns that currently exist in the instructor dashboard (the buttons make AJAX requests instead of submitting a general HTML form, etc.), but I tried to be careful not to overdo it with optimizing legacy code.

A few key files from the dogwood release that contained the legacy dashboard functionality:
- https://github.com/mitodl/edx-platform/blob/mitx/dogwood.3-3/lms/templates/courseware/legacy_instructor_dashboard.html
- https://github.com/mitodl/edx-platform/blob/mitx/dogwood.3-3/lms/djangoapps/instructor/views/legacy.py#L211

### CODE REVIEW

File: conf/locale/en/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -6125,6 +6125,10 @@ msgid "Data Download"
 msgstr ""
 
 #: lms/djangoapps/instructor/views/instructor_dashboard.py
+msgid "Export Grades"
+msgstr ""
+
+#: lms/djangoapps/instructor/views/instructor_dashboard.py
```
Comment (pdpinch) [inline conf/locale/en/LC_MESSAGES/django.po]: are changes to the .po files necessary?

File: lms/djangoapps/courseware/access.py [hunk 2]
```diff
@@ -314,6 +314,10 @@ def _can_enroll_courselike(user, courselike):
     # which actually points to a CourseKey. Sigh.
     course_key = courselike.id
 
+    if settings.FEATURES.get('RESTRICT_ENROLL_NO_ATSIGN_USERNAMES') and '@' in user.username:
+        log.warning("ENROLLMENT DENIED in %s because @ in username %s" % (course_key, user.username))
+        return ACCESS_DENIED
+
```
Comment (pdpinch) [inline lms/djangoapps/courseware/access.py]: This seems unrelated, but we might have picked it up from a cherry-pick. Tobias may know more.

**Review by pdpinch — COMMENTED** [review]
Still doing functional tests, but I had a couple of comments / questions.

File: lms/djangoapps/courseware/access.py (hunk 2 above)
Comment (blarghmatey) [reply lms/djangoapps/courseware/access.py]: Yeah, this is a change that is unrelated to this PR. It is most likely an artifact of the branch that this was based off of.

File: conf/locale/en/LC_MESSAGES/django.po (hunk 1 above)
Comment (gsidebo) [reply conf/locale/en/LC_MESSAGES/django.po]: ...honestly i have no idea how these got into the commit in the first place. `.gitignore` contains the pattern `*.po*`. i'll get rid of these.

Comment (gsidebo) [reply conf/locale/en/LC_MESSAGES/django.po]: looks like rebasing on `ficus-1rc1` again fixed it 👍

**Comment by gsidebo** [comment]
Closing in favor of #18

**Pull request closed without merging**